src/main/java/com/yugabyte/sample/apps/SqlForeignKeysAndJoins.java

```diff
@@ -93,7 +95,16 @@ public void dropTable() throws Exception {
 
   @Override
   public void createTablesIfNeeded(TableOp tableOp) throws Exception {
-    Connection connection = getPostgresConnection();
+    Connection connection = null;
+    do {
+      try {
+        connection = getPostgresConnection();
+        break;
+      } catch (Exception e) {
+        if (!SqlInserts.sleep(sleepIntervalMs)) return;
```

Comment:
`SqlInserts.sleep()`, sleep function should be move AppBase as well. It doesn't make sense to use `SqlInserts` function in `SqlForeignKeysAndJoins`.